ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/streaming/StreamedQueryResourceTest.java

```diff
@@ -471,6 +523,29 @@ public void shouldReachConcurrentLimit() {
     assertThat(exception.getValue().getMessage(), containsString("concurrencyLimiter Error!"));
   }
 
+  @Test
+  public void shouldReachConcurrentLimitStreamPullQuery() {
+    // Given:
+    when(rateLimiter.tryAcquire()).thenReturn(true);
+    when(concurrencyLimiter.increment()).thenThrow(new KsqlException("concurrencyLimiter Error!"));
+    when(mockDataSource.getDataSourceType()).thenReturn(DataSourceType.KSTREAM);
+
+    // When:
+    final EndpointResponse response =
+        testResource.streamQuery(
+            securityContext,
+            new KsqlRequest(PULL_QUERY_STRING, Collections.emptyMap(), Collections.emptyMap(), null),
+            new CompletableFuture<>(),
+            Optional.empty(),
+            KsqlMediaType.LATEST_FORMAT,
+            new MetricsCallbackHolder(),
+            context);
+
+    // Then:
+    assertThat(response.getStatus(), is(500));
```

Comment:
It looks like we have already been returning a 500 for limit violations, so this isn't a problem with your PR specifically, but just to point it out, 500 (server error) is the wrong response code. It should be a 4xx (bad request) type error. 429 is probably the best error, though it's not a perfect match for concurrency violations: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429

Reply:
Hmm, I am not sure where the error codes get assigned in the code base. I will open a ticket to fix this as part of the general error code fixing initiative we will do